contracts/SnappAuction.sol

```diff
@@ -249,6 +252,69 @@ contract SnappAuction is SnappBase {
         auctions[slot].tentativeState = proposedStateRoot;
     }
 
+    function winnerApplyAuction(
+        uint slot,
+        bytes memory pricesAndVolumes
+    )
+        public onlyOwner()
+    {
+        require(auctions[slot].appliedAccountStateIndex == 0, "Auction already applied");
+        require(
+            block.timestamp > biddingStartTime(slot) + 3 minutes,
+            "Requested auction still in bidding phase."
+        );
+        require(
+            block.timestamp < biddingStartTime(slot) + 270 seconds,  // 3 + 1.5 minutes
+            "Too late for winning result to submit solution"
+        );
+        require(
+            auctions[slot].solver == msg.sender,
+            "Only winner of bidding phase may apply auction here"
+        );
+        internalApplyAuction(slot, auctions[slot].tentativeState, pricesAndVolumes);
+    }
+
+    function fallbackApplyAuction(
+        uint slot,
+        bytes32 _currStateRoot,
+        bytes32 newStateRoot,
+        bytes memory pricesAndVolumes
+    )
+        public onlyOwner()
+    {
+        require(slot != MAX_UINT && slot <= auctionIndex, "Requested auction slot does not exist");
+        require(slot == 0 || auctions[slot-1].appliedAccountStateIndex != 0, "Must apply auction slots in order!");
+        require(auctions[slot].appliedAccountStateIndex == 0, "Auction already applied");
+        require(coreData.stateRoots[stateIndex()] == _currStateRoot, "Incorrect state root");
+
+        require(
+            block.timestamp > biddingStartTime(slot) + 270 seconds,  // 3 + 1.5 minutes
+            "Waiting on winner to provide solution"
+        );
+        require(
+            block.timestamp < biddingStartTime(slot) + 6 minutes,
+            "Waiting on winner to provide solution"
+        );
+        // TODO - Deal with this part of the settlement period.
+        // Assuming we take any provided solution and apply it!?
+        internalApplyAuction(slot, newStateRoot, pricesAndVolumes);
+    }
+
+    function nudgeTrivialAuction(uint slot) public {
```

Comment:
Who would call this function? Should it be triggered by another function, e.g. placing an order could check whether it needs to be triggered.

Reply:
This would have been called by anyone who wants to clear an auction that hasn't been cleared yet. However, this is in the process of being reworked.